core/pkg/server/sender.go

```diff
@@ -1134,6 +1136,54 @@ func (s *Sender) sendSync(record *service.Record, request *service.SyncRequest)
 	s.fwdChan <- rec
 }
 
+func (s *Sender) sendStopStatus(record *service.Record, _ *service.StopStatusRequest) {
+	// TODO: ideally we could get it direclty from the settings
+	entity := s.RunRecord.GetEntity()
+	project := s.RunRecord.GetProject()
+	runId := s.RunRecord.GetRunId()
```

Comment:
is it not available on the settings? my new `runfiles` code uses it and seems to work fine, but I can change it if it's not reliable

Reply:
i think it should be, we re-assign it, not sure about runId... it is just not consistent in our sender code.
Ideally i would want to get rid of this and just have your new settings...